### PR TITLE
Code block syntax highlighting

### PR DESCRIPTION
**Pull request by ClaytonPassmore**
## TL;DR
- Added syntax highlighting to code blocks using `highlight.js`.
- Updated the README based on the issues I ran into during installation.

## Reason for change
I write in Markdown quite often for work. Usually when I do, I include code snippets in my documents. Being able to preview my documents with syntax highlighting in the code blocks is very high value for me (and hopefully others).

**Full disclaimer**: I have never written vala or used meson before, so hopefully this PR isn't too much of a mess :stuck_out_tongue: 

## highlight.js
highlight.js is built for exactly this use case. You can find more info:
- On their [website](https://highlightjs.org/)
- On their [GitHub repo](https://github.com/isagalaev/highlight.js/)

## Screenshots
What's a PR without some screenshots?

You can specify the code language type after the triple tick like so:
screenshot from 2017-12-19 20 06 56

This results in some nice syntax highlighting in the preview:
screenshot from 2017-12-19 20 07 16

Things are similar in the dark theme:
screenshot from 2017-12-19 20 07 40

And in the dark preview:
screenshot from 2017-12-19 20 07 30

If you don't specify a language, it will not highlight the block:
screenshot from 2017-12-19 20 09 22

### CODE REVIEW

**Comment by lainsce** [comment]
This is perfectly valid Vala and Meson. Thanks for this massive PR! :heart:

**Comment by ClaytonPassmore** [comment]
@lainsce no problem! I'm happy to contribute 😄
